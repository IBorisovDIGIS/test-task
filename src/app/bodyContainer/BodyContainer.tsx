import React, { ReactElement } from 'react';
import {
  Container,
  Grid,
  Typography,
  makeStyles,
} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  appBarSpacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    height: '100vh',
    overflow: 'auto',
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
  },
  title: {
    fontSize: 39,
    fontWeight: 300,
  },
}));

interface IBody {
    children: ReactElement;
    title: string;
}

const BodyContainer: React.FC<IBody> = ({ children, title }:IBody) => {
  const classes = useStyles();

  return (
    <main className={classes.content}>
      <div className={classes.appBarSpacer} />
      <Container maxWidth="lg" className={classes.container}>
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <Typography component="h1" variant="h3" className={classes.title}>
              {title}
            </Typography>
          </Grid>
          <Grid item xs={12} id="body-wrapper">
            {children}
          </Grid>
        </Grid>
      </Container>
    </main>
  );
};

export default BodyContainer;
