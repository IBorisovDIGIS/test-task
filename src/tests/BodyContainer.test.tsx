// CustomIcon.tsx test
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Container, Grid, Typography } from '@material-ui/core';
import toJson from 'enzyme-to-json';
import BodyContainer  from '../app/bodyContainer/BodyContainer';

configure({ adapter: new Adapter() });

describe('Body Container should render correctly', () => {
  const TestChild = <div>Test</div>;

  it('Should renders correctly', () => {
    const container = shallow(
      <BodyContainer title="Title">
        {TestChild}
      </BodyContainer>,
    );

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should render components and display title correctly', () => {
    const container = shallow(
      <BodyContainer title="Title">
        {TestChild}
      </BodyContainer>,
    );

    expect(Container).toBeDefined();
    expect(Grid).toBeDefined();
    expect(Typography).toBeDefined();
    const title = container.find(Typography);
    expect(title.text()).toEqual('Title');
  });

  it('Should render children correctly', () => {
    const container = shallow(
      <BodyContainer title="Title">
        {TestChild}
      </BodyContainer>,
    );
    const { children } = container.find('#body-wrapper').props();
    expect(children).toEqual(TestChild);
  });
});
