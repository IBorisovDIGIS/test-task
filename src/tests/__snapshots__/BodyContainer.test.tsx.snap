// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Container should render correctly Should renders correctly 1`] = `
<main
  className="makeStyles-content-2"
>
  <div
    className="makeStyles-appBarSpacer-1"
  />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-3"
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-4"
          component="h1"
          variant="h3"
        >
          Title
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        id="body-wrapper"
        item={true}
        xs={12}
      >
        <div>
          Test
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</main>
`;
